2.0-PROPOSAL/README.md

```diff
@@ -0,0 +1,885 @@
+proposal introduces numerous architectural complications and is therefore not
+entertained by the Brigade 2.0 proposal. The philosophical pivot to "let workers
+work" is in no small measure intended to reduce event handling-latency for
+simple use cases that can not abide a worker pod starting _only_ to launch one
+other pod.
+
+Refer to:
+
+* [Issue #977](https://github.com/brigadecore/brigade/issues/977)
+
+## Proposed Path Forward
+
+### Guiding Principles
+
+The author proposes several guiding principles for the design and development of
+Brigade 2.0:
+
+1. Neither Kubernetes expertise nor cluster credentials must be a prerequisite
+   for success.
+    1. Kubernetes is neither to be removed nor 100% abstracted. Realistic goals
+       are to minimize Kubernetes' use as anything other than a substrate for
+       workload execution and to isolate Brigade's touchpoints with Kubernetes
+       to specific components.
+    1. Some degree of freedom to "drop down" to Kubernetes may be retained to
+       enable implementation of advanced / fringe use cases, but the need to
+       utilize this should be _exceptional._
+
+1. The UX must be both simplified and improved while striving to remain
+   recognizable and comfortable for experienced Brigade users.
+
+1. Security and reliability must be improved beyond the Brigade 1.x status quo.
+
+1. Product evaluation (aka "tire kicking") / local use for development must not
+   become onerous.
+    1. For any new infrastructure dependencies (for instance, a data store or
+       message bus), a default Brigade installation must provision in-cluster
+       instances running in modest configurations that prioritize minimized
+       resource use. Documentation may offer guidance on optimizing
+       installations for production use.
+    1. Security improvements must not encumber evaluation.
+
+1. Small compromises are deemed acceptable if they remediate larger issues. For
+   instance, logs that stream with greater latency than Brigade 1.x's are
+   acceptable if they're less susceptible to data loss.
+
+1. Design cues should be taken from Kubernetes (or other familiar sources)
+   wherever possible in order to produce architecture, tooling, APIs, etc. that
+   feel familiar to both contributors and users and incur minimal learning
+   curve.
+
+### Domain Model
+
+Incidental types required to address non-functional requirements (such as access
+control) notwithstanding, Brigade 2.0 is proposed to deal with only two "top
+level" domain elements: projects and events.
+
+__Projects__ pair _subscriptions_ to various events with a _template_ for a
+worker that will handle such events. Projects additionally play a role similar
+to that of a Kubernetes namespace in that they define a logical scope for access
+controls. For instance, rather than granting a user permissions to list or view
+_all_ events in Brigade, a user might be granted permissions to list or view
+events _for a particular project_.
```

Comment:
From @brandonkal:

> The term Project can be confusing and you should consider renaming it to something else. Because it is really not a project but an isolated config for launching workers and storing secrets. Perhaps something like Dispatcher or Subscription.

This is a legacy of Brigade 1.x and it is a stated goal to make 2.0 feel familiar to existing Brigade users, despite the many changes.

imho, "project" is also colloquially understood across many different contexts in our industry to denote a unit of isolation or independence. Specifically wrt Brigade 1.x, I've never heard of any prior confusion over the use of this term. (Contrast this with "builds," and the frequent complaint, "but I'm not _building_ anything in my use case.")

Reply:
The issue is that when I think of project, I consider the code to be part of that project. But in brigade terms a Project is only a subset of that. It's more of something that binds your project to brigade.
Consider a user starting out with brigade:
developer: I wish to start a new project. Let's create a repo to store my code.
developer: I've heard about brigade, let's try hooking up that.
The developer reads the docs, where it suggests storing config in a separate repo. I love that these can be separate but it is a bit confusing when learning brigade the first time.
developer: So I need to create another project just to configure brigade?

And that is the issue. Because many consider the code repository the project's home, it is odd having another project somewhere else.
Example confusion: https://youtu.be/hreCyC7XnOw?t=1848
It's not that bad once you get the hang of it but 2.0 would be an opportunity to give it a more descriptive name.

Even a change to ProjectConfig or Pipeline would help imho.

Reply:
You're arbitrarily subscribing to a narrow definition of "project" as being equivalent to "repository." Even within our industry, it's not difficult to find many other uses of the word that have nothing to do with a source code repository. 

It seems that some minor fixes to the docs (which would be completely overhauled for 2.0 anyway) would remediate the sort of confusion Joe Beda seems to have in that clip.

Reply:
Objections are noted. I've spoken with two other maintainers offline and we agree that unless a majority of maintainers (who _all_ are reviewing this proposal and community feedback carefully) voice opposition to the term "project," the term will stand.